### DISCUSSION TITLE
StepFunction: How to add a date run condition on the parallel state of State Machine

### DISCUSSION THREAD

**Discussion by sandycoolsash** [aws/aws-cdk#18250] [General]
Hi 

I have state machine which has a bunch of task and parallel state. I have my state machine schedule to run daily but i want my parallel state to only start when its 22nd of the month and rest all days of the month the parallel state should skip/not run.

Any idea/suggestion how i can get this implemented
![state machine date run condition](https://user-images.githubusercontent.com/56182724/147967990-ecd1cb91-7a62-47bd-a9bf-4e59afca045f.jpg)

ASL Definition
{
  "StartAt": "Choice State Name?",
  "States": {
    "Choice State Name?": {
      "Type": "Choice",
      "Choices": [
        {
          "Variable": "$.StateName",
          "IsPresent": false,
          "Next": "Daily_Job_1"
        },
        {
          "Variable": "$.StateName",
          "StringEquals": "Daily_Job_1",
          "Next": "Daily_Job_1"
        },
        {
          "Variable": "$.StateName",
          "StringEquals": "Parallel_Chain",
          "Next": "Parallel_Chain"
        },
        {
          "Variable": "$.StateName",
          "StringEquals": "Daily_Job_3",
          "Next": "Daily_Job_3"
        }
      ],
      "Default": "Job Failed"
    },
    "Job Failed": {
      "Type": "Fail",
      "Error": "!!! Start Job Name passed is not correct and not matched the defined rule",
      "Cause": "Rule not matched for Start Job Name"
    },
    "Daily_Job_1": {
      "Next": "Parallel_Chain",
      "Type": "Task",
      "ResultPath": null,
      "Resource": "arn:aws:states:::batch:submitJob.sync",
      "Parameters": {
        "JobDefinition": "arn:aws:batch:us-west-2:XXXXX:job-definition/dataservices-batch-job-definition-poc:1",
        "JobName": "Daily_Job_1",
        "JobQueue": "arn:aws:batch:us-west-2:XXXXX:job-queue/getting-started-job-queue-batch-poc",
        "ContainerOverrides": {}
      }
    },
    "Parallel_Chain": {
      "Type": "Parallel",
      "ResultPath": null,
      "Next": "Daily_Job_3",
      "Branches": [
        {
          "StartAt": "Daily_Job_2a",
          "States": {
            "Daily_Job_2a": {
              "Next": "Daily_Job_2b",
              "Type": "Task",
              "ResultPath": null,
              "Resource": "arn:aws:states:::batch:submitJob.sync",
              "Parameters": {
                "JobDefinition": "arn:aws:batch:us-west-2:XXXXX:job-definition/dataservices-batch-job-definition-poc:1",
                "JobName": "Daily_Job_2a",
                "JobQueue": "arn:aws:batch:us-west-2:XXXXX:job-queue/getting-started-job-queue-batch-poc",
                "ContainerOverrides": {}
              }
            },
            "Daily_Job_2b": {
              "End": true,
              "Type": "Task",
              "ResultPath": null,
              "Resource": "arn:aws:states:::batch:submitJob.sync",
              "Parameters": {
                "JobDefinition": "arn:aws:batch:us-west-2:XXXXX:job-definition/dataservices-batch-job-definition-poc:1",
                "JobName": "Daily_Job_2b",
                "JobQueue": "arn:aws:batch:us-west-2:XXXXX:job-queue/getting-started-job-queue-batch-poc",
                "ContainerOverrides": {}
              }
            }
          }
        },
        {
          "StartAt": "Daily_Job_2c",
          "States": {
            "Daily_Job_2c": {
              "Next": "Daily_Job_2d",
              "Type": "Task",
              "ResultPath": null,
              "Resource": "arn:aws:states:::batch:submitJob.sync",
              "Parameters": {
                "JobDefinition": "arn:aws:batch:us-west-2:XXXXX:job-definition/dataservices-batch-job-definition-poc:1",
                "JobName": "Daily_Job_2c",
                "JobQueue": "arn:aws:batch:us-west-2:XXXXX:job-queue/getting-started-job-queue-batch-poc",
                "ContainerOverrides": {}
              }
            },
            "Daily_Job_2d": {
              "Next": "Daily_Job_2e",
              "Type": "Task",
              "ResultPath": null,
              "Resource": "arn:aws:states:::batch:submitJob.sync",
              "Parameters": {
                "JobDefinition": "arn:aws:batch:us-west-2:XXXXX:job-definition/dataservices-batch-job-definition-poc:1",
                "JobName": "Daily_Job_2d",
                "JobQueue": "arn:aws:batch:us-west-2:XXXXX:job-queue/getting-started-job-queue-batch-poc",
                "ContainerOverrides": {}
              }
            },
            "Daily_Job_2e": {
              "End": true,
              "Type": "Task",
              "ResultPath": null,
              "Resource": "arn:aws:states:::batch:submitJob.sync",
              "Parameters": {
                "JobDefinition": "arn:aws:batch:us-west-2:XXXXX:job-definition/dataservices-batch-job-definition-poc:1",
                "JobName": "Daily_Job_2e",
                "JobQueue": "arn:aws:batch:us-west-2:XXXXX:job-queue/getting-started-job-queue-batch-poc",
                "ContainerOverrides": {}
              }
            }
          }
        }
      ]
    },
    "Daily_Job_3": {
      "End": true,
      "Type": "Task",
      "ResultPath": null,
      "Resource": "arn:aws:states:::batch:submitJob.sync",
      "Parameters": {
        "JobDefinition": "arn:aws:batch:us-west-2:XXXXX:job-definition/dataservices-batch-job-definition-poc:1",
        "JobName": "Daily_Job_3",
        "JobQueue": "arn:aws:batch:us-west-2:XXXXX:job-queue/getting-started-job-queue-batch-poc",
        "ContainerOverrides": {}
      }
    }
  },
  "TimeoutSeconds": 7200
}

**Comment by indrora** [comment]
This is a question best asked of the state function team, not the CDK.
